canvas_zoom_meetings.py

```diff
@@ -0,0 +1,92 @@
+# Script to get all sites where Zoom is visible and retrieve the meetings to generate a report
+
+import json
+import re
+import os
+
+import requests
+import logging
+
+from canvasapi import Canvas
+from canvasapi.exceptions import ResourceDoesNotExist
+from collections import OrderedDict
+from bs4 import BeautifulSoup as bs
+from urllib.parse import urljoin
+
+# read configurations
+try:
+    with open(os.path.join('config', 'env.json')) as env_file:
+        ENV = json.loads(env_file.read())
+except FileNotFoundError:
+    logger.error(
+        'Configuration file could not be found; please add env.json to the config directory.')
+
+LOG_LEVEL = ENV.get('LOG_LEVEL', 'DEBUG')
+logging.basicConfig(level=LOG_LEVEL)
+
+# These two lines enable debugging at httplib level (requests->urllib3->http.client)
+# You will see the REQUEST, including HEADERS and DATA, and RESPONSE with HEADERS but without DATA.
+# The only thing missing will be the response.body which is not logged.
+import http.client as http_client
+if LOG_LEVEL == logging.DEBUG:
+    http_client.HTTPConnection.debuglevel = 1
+
+# You must initialize logging, otherwise you'll not see debug output.
+logging.basicConfig(level=LOG_LEVEL)
+logger = logging.getLogger(__name__)
+logger.setLevel(LOG_LEVEL)
+
+requests_log = logging.getLogger("requests.packages.urllib3")
+requests_log.setLevel(LOG_LEVEL)
+requests_log.propagate = True
+
+#sites_to_update = [109190]
+
+CANVAS = Canvas(ENV.get("CANVAS_API_URL"), ENV.get("CANVAS_API_KEY"))
+
+account = CANVAS.get_account(ENV.get("CANVAS_ACCOUNT" ,1))
+
+courses = account.get_courses(enrollment_term_id=ENV.get("CANVAS_TERM", 1))
+for course in courses:
+    logger.info("Fetching course %s", course)
+    # Get tabs and look for zoom
```

Comment:
Only for the published courses:

+    # only proceed when course is published
+    if course.workflow_state != 'available':
+        continue
+

Reply:
Yeah, we can just call get_courses with the published [kwarg](https://canvas.instructure.com/doc/api/accounts.html#method.accounts.courses_api) forgot to add that.